common/network-common/src/main/java/org/apache/spark/network/buffer/FileSegmentManagedBuffer.java

```diff
@@ -37,13 +37,24 @@
  * A {@link ManagedBuffer} backed by a segment in a file.
  */
 public final class FileSegmentManagedBuffer extends ManagedBuffer {
-  private final TransportConf conf;
+  private final boolean lazyFileDescriptor;
+  private final int memoryMapBytes;
   private final File file;
   private final long offset;
   private final long length;
 
   public FileSegmentManagedBuffer(TransportConf conf, File file, long offset, long length) {
-    this.conf = conf;
+    this(conf.lazyFileDescriptor(), conf.memoryMapBytes(), file, offset, length);
+  }
+
+  public FileSegmentManagedBuffer(
```

Comment:
Like the following code?

```java
  public FileSegmentManagedBuffer(TransportConf conf, File file, long offset, long length) {
    this.lazyFileDescriptor = conf.lazyFileDescriptor();
    this.memoryMapBytes = conf.memoryMapBytes();
    this.file = file;
    this.offset = offset;
    this.length = length;
  }
```

the code `conf.lazyFileDescriptor();` or `conf.memoryMapBytes();`  creates a NoSuchElementException instance.